src/main/java/org/truffleruby/core/array/ArrayNodes.java

```diff
@@ -733,7 +733,17 @@ public Object deleteAt(DynamicObject array, int index,
     @ImportStatic(ArrayGuards.class)
     public abstract static class EachNode extends YieldingCoreMethodNode {
 
-        @Specialization(guards = "strategy.matches(array)", limit = "ARRAY_STRATEGIES")
+        @Specialization(guards = { "strategy.matches(array)", "strategy.getSize(array) == 1" }, limit = "ARRAY_STRATEGIES")
+        public Object eachOne(DynamicObject array, DynamicObject block,
+                              @Cached("of(array)") ArrayStrategy strategy) {
+            final ArrayMirror store = strategy.newMirror(array);
+
+            yield(block, store.get(0));
+
+            return array;
```

Comment:
So much vertical space :wink: (but do as you like)

Reply:
Setup -> work -> return. That's the logical breakdown for me at least. But I'm seemingly in the minority so I can collapse that.